rest/data/host_create.go

```diff
@@ -343,34 +372,15 @@ func makeDockerIntentHost(ctx context.Context, env evergreen.Environment, taskID
 
 }
 
-func makeEC2IntentHost(ctx context.Context, env evergreen.Environment, taskID, userID, publicKey string, createHost apimodels.CreateHost) (*host.Host, error) {
+func makeEC2IntentHost(ctx context.Context, env evergreen.Environment, taskID, userID, publicKey string, createHost apimodels.CreateHost, d distro.Distro) (*host.Host, error) {
 	if createHost.Region == "" {
 		createHost.Region = evergreen.DefaultEC2Region
 	}
-	// get distro if it is set
-	d := distro.Distro{}
 	ec2Settings := cloud.EC2ProviderSettings{}
 	var err error
```

Comment:
nit: it shouldn't be necessary to pre-define `err` anymore since the `if createHost.Distro != ""` block doesn't need it.